### PR TITLE
Fixed bug where MaximumRespawnAmount is always 15

### PR DESCRIPTION
**Pull request by RinTheWeeb**
Because of

` private void ReissueNextKnownTeam()
        {
            var team = SpawnableTeam;
            if (team != null)
            {
                MaximumRespawnAmount = team.Value.MaxWaveSize;
            }
}`

`
 set
            {
                nextKnownTeam = value;
                ReissueNextKnownTeam();
            }
`

And added some extra checks

### CODE REVIEW

File: Exiled.Events/Patches/Events/Server/RespawningTeam.cs [hunk 1]
```diff
@@ -46,16 +46,9 @@ private static bool Prefix(RespawnManager __instance)
                             .Where(p => p.IsDead && !p.IsOverwatchEnabled));
 
                     if (__instance._prioritySpawn)
-                    {
-                        var tempList = ListPool<API.Features.Player>.Shared.Rent();
-                        tempList.AddRange(list.OrderBy(item => item.ReferenceHub.characterClassManager.DeathTime));
-                        ListPool<API.Features.Player>.Shared.Return(list);
-                        list = tempList;
-                    }
+                        list.OrderBy(item => item.ReferenceHub.characterClassManager.DeathTime).ToList();
```
Comment (iRebbok) [inline Exiled.Events/Patches/Events/Server/RespawningTeam.cs:49]: as far as I can tell, ToList() creates a new list and that list isn't assigned to the `list` variable

**Comment by iRebbok** [comment]
IDK at all, it should work like setting `MaximumRespawnAmount` after `NextKnowTeam`

File: Exiled.Events/Patches/Events/Server/RespawningTeam.cs (hunk 1 above)
Comment (RinTheWeeb) [reply Exiled.Events/Patches/Events/Server/RespawningTeam.cs]: Fixed

**Comment by RinTheWeeb** [comment]
> 
> 
> IDK at all, it should work like setting `MaximumRespawnAmount` after `NextKnownTeam`

After the line Handlers.Server.OnRespawningTeam(ev);

The limit got reset back to the default value `team.Value.MaxWaveSize` instead of keeping its original value. 

That was because of the of that function you made when ev.Spawnable is getted? (I think).

I wasn't 100% which part of that code caused it, but this resolved it at least.

**Comment by iRebbok** [comment]
Ah, then it requires another fix, I'll close this PR once I get it to work correctly

**Comment by iRebbok** [comment]
Fixed in development with another implementation

**Pull request closed without merging**